lib/commands/src/token.rs

```diff
@@ -0,0 +1,84 @@
+#[derive(Debug, Clone, Eq, Hash, PartialEq)]
+pub enum Token {
+    /// Token used to represent a finished command (i.e. no more parameters required)
+    // todo: this is likely not the right way to represent this
+    Empty,
+
+    /// A bot-defined value ("member" in `pk;member MyName`)
+    Value(Vec<String>),
+    /// A command defined by multiple values
+    // todo!
+    MultiValue(Vec<Vec<String>>),
+
+    FullString,
+
+    /// Member reference (hid or member name)
+    MemberRef,
+    MemberPrivacyTarget,
+
+    PrivacyLevel,
+
+    // currently not included in command definitions
+    // todo: flags with values
+    Flag,
+}
+
+pub enum TokenMatchResult {
+    NoMatch,
+    /// Token matched, optionally with a value.
+    Match(Option<String>),
+}
+
+// move this somewhere else
+lazy_static::lazy_static!(
+    static ref MEMBER_PRIVACY_TARGETS: Vec<String> = vec![
```

Comment:
spitballing: should privacy targets etc be macro-generated enums, with a token type for referencing them in the command structure DSL? even if they just get pushed through FFI as a string argument in the end, it'd make the whole "self-documenting" goal easier